Replace fat cli with a light one

diegosalvi opened the pull request:
This is a draft to create a slim cli while keeping "fat" cli artifact

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

eolivelli reviewed (changes requested):
Makes sense to me

It is quite hard for the user to understand why have two CLI versions.

I guess that the problem is about the fact that by default from the CLI you can start the embedded server. 
I don't think that this is something that is expected.
I would simply fix the original CLI.
if you need to start a server over existing data you can download the zip distribution

aluccaroni commented:
+1 to simplify the cli and keep only one, even if it's a breaking change

eolivelli reviewed (changes requested):
As @aluccaroni said, it is better to have only one CLI tool and not two.
It would be confusing to have two packages

diegosalvi commented:
Sounds good to me

diegosalvi commented:
After tests this could be merged

eolivelli commented on herddb-cli/pom.xml:
```diff
@@ -59,31 +59,6 @@
     </dependencies>
     <build>
         <plugins>
-            <plugin>
```
I think that if you drop this plugin the .jar won't be able to work.

We still need a uber jar, but we can remove anything that is not required to run the CLI.
I think that most of the weight is due to the rocksdb jar.
I would drop useless jars, like we did with the "thin" JDBC driver

diegosalvi replied:
The shaded jar generate various problems with his shaded dependencies.
You can just download the services and use the cli with it or in a more complex project run it with your other libraries.
Shading every dependency has many classloading problems when you want to upgrade some dependency (and then for some "strange" behavior you find loaded old classes from the shaded lib)

eolivelli replied:
The Uber jar is not intended to be used as dependency but only to run the CLI from the command line.

I agree that we can let users use the zip file.
Also, now that we have docker images, it is super easy to run docker run -it .....

eolivelli reviewed (commented):
Do we have test Cades that very that the CLI packaged in the zip distro works?

Did you try it manually?

diegosalvi commented:
Good catch! I forgot the cli shell and bat files in services.
Fixed services package:
* changed scripts to use java -cp instead of java -jar
* moved herddb-cli to lib instead of main directory (it's cleaner this way)

dmercuriali reviewed (approved):
Tested locally. LGTM

eolivelli reviewed (approved):
LGTM
nice work !
